python/cudf/cudf/core/column/string.py

```diff
@@ -1663,6 +1663,78 @@ def match(self, pat, case=True, flags=0, **kwargs):
             cpp_match_re(self._column, pat), **kwargs
         )
 
+    def url_decode(self, **kwargs):
+        """
+        Returns a URL-decoded format of each string.
+        No format checking is performed. All characters
+        are expected to be encoded as UTF-8 hex values.
+
+        Returns
+        -------
+        Series or Index.
+
+        """
+
+        return self._return_or_inplace(cpp_url_decode(self._column), **kwargs)
+
+    def url_encode(self, **kwargs):
+        """
+        Returns a URL-encoded format of each string.
+        No format checking is performed.
+        All characters are encoded except for ASCII letters,
+        digits, and these characters: ‘.’,’_’,’-‘,’~’.
+        Encoding converts to hex using UTF-8 encoded bytes.
+
+        Returns
+        -------
+        Series or Index.
+
+        """
+        return self._return_or_inplace(cpp_url_encode(self._column), **kwargs)
+
+    def code_points(self, **kwargs):
+        """
+        Returns an array by filling it with the UTF-8 code point
+        values for each character of each string.
+        This function uses the len() method to determine
+        the size of each sub-array of integers.
+
+        Returns
+        -------
+        Series or Index.
+        """
+        from cudf.core.series import Series, Index
+
+        new_col = cpp_code_points(self._column)
+        if self._parent is None:
+            return new_col
+        elif isinstance(self._parent, Series):
+            return Series(new_col, name=self._parent.name)
+        elif isinstance(self._parent, Index):
+            return column.as_index(new_col, name=self._parent.index.name)
```

Comment:
`self._parent` is already an Index here
```suggestion
            return column.as_index(new_col, name=self._parent.name)
```